fhevm/tee_bit.go

```diff
@@ -0,0 +1,145 @@
+package fhevm
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teeShlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
+		case 1:
+			return uint64((uint8(a) << uint8(b)))
+		case 2:
+			return uint64((uint8(a) << uint8(b)))
+		case 3:
+			return uint64((uint16(a) << uint16(b)))
+		case 4:
+			return uint64((uint32(a) << uint32(b)))
+		case 5:
+			return (a << b)
```

Comment:
Once we do uint128, we will anyways need a bytes shifter/rotator. So I would argue to use *big.Int directly, and remove those switch cases.

Reply:
Is the result of `uint8(a) << uint8(b)` the same as `big.Int(a) << big.Int(b)`?

Reply:
The result of other operations is same in case of uint and big.int? I'm not sure about this.
In zama code, they didn't implement operations with uint128 and uint160 yet.

Reply:
Yeah, once we do uint128, I think big.Int doesn't have << and >> support. We might need to convert to bytes, and do the operations on bytes

Let's keep it like this for now